lib/Bort.pm

```diff
@@ -371,7 +371,14 @@ sub user_name    { shift; my $user = $_[-1]; $user_names{$user} // () }
 
 sub my_user_name    { $my_user_name }
 sub channel_by_name { $channel_by_name{$_[-1]} // () };
-sub user_by_name    { $user_by_name{$_[-1]} // () };
+sub user_by_name    {
+  # Sometimes we get a user_id slack link instead,
+  #   so we just return the sanitised user_id in that case
+  ($_[-1] =~ s/^<@(U[A-Z0-9]{8})>/$1/)
```

Comment:
No good. `@_` is aliased to the arg list, so this either modifies the original arg, or crashes if its immutable (ie a literal).

Either use `s///r` (non-destructive replacement) or (my preference) a capture. But we're starting to get pretty complicated, so maybe just write the method out in full (untested):

```perl
sub user_by_name {
  my $name = pop @_;
  return $1 if $name =~ m/^<@(U[A-Z][0-9]{8})>/;
  $user_by_name{$_[-1]} // ()
}
```

Reply:
Ahhh, didn't think about immutable case.

My first cut looked more like that, before I tried being *clever*